### PR TITLE
Refactored and improved VTK model and pane [doc-build]

### PR DESCRIPTION
**Pull request by philippjfr**
Does some refactoring for the VTK pane and adds some features:

* VTK pane now supports responsive sizing modes
* VTK pane now supports syncing (and therefore linking) camera state

@xavArtley, review would be appreciated but I'll probably merge some time tomorrow in any case.

@dharhas, it currently only supports uni-directional links between objects, which is something I need to think about and address in general, but you could now link multiple plots together and then interact with the plot they are linked to:

vtk_link

### CODE REVIEW

**Comment by xavArtley** [comment]
In notebook the vtk panel does not appear for me
image
If I use the show metho it seems to work
image

**Comment by philippjfr** [comment]
Is that JupyterLab or classic notebook? I can't reproduce issues in either.

**Comment by xavArtley** [comment]
jupyter lab

**Comment by philippjfr** [comment]
Ah, have you updated the JupyterLab extension recently? We made a major change to the way output is published in the notebook.

**Comment by philippjfr** [comment]
You'll need:

@pyviz/jupyterlab_pyviz v0.7.1

**Comment by xavArtley** [comment]
Ok greats it works now.

**Comment by philippjfr** [comment]
Actually bi-directional linking works okay, and there's no need for the trigger:

```
vtk = pn.pane.VTK('https://raw.githubusercontent.com/Kitware/vtk-js/master/Data/StanfordDragon.vtkjs',
                  sizing_mode='stretch_both')
vtk2 = pn.pane.VTK('https://raw.githubusercontent.com/Kitware/vtk-js/master/Data/StanfordDragon.vtkjs',
                   sizing_mode='stretch_both')

vtk.jslink(vtk2, camera='camera')
vtk2.jslink(vtk, camera='camera')

pn.Row(vtk, vtk2)
```

I'll probably add a bi-directional flag to the ``jslink`` method.

**Comment by philippjfr** [comment]
@xavArtley Finally got vtk to run on Travis so we'll be able to add some more tests.

File: panel/models/vtk.ts [hunk 1]
```diff
@@ -1,60 +1,90 @@
...
+  _get_camera_state(): void {
+    if (!this._setting) {
+      this._setting = true;
+      const state = clone(this._camera.get());
+      delete state.classHierarchy;
+      delete state.vtkObject;
+      delete state.vtkCamera;
+      delete state.viewPlaneNormal;
+      this.model.camera = state;
+      this.model.properties.camera.change.emit();
```
Comment (xavArtley) [inline panel/models/vtk.ts]: Is it really necessary to emit change explicitly?
As we set a new value to model.camera, I would excpect the change event would be already emit

Comment (philippjfr) [reply panel/models/vtk.ts]: I think you're right, thanks!

**Pull request merged**